### ISSUE TITLE
"max_vertices = specialization constant" should generate an error?

### ISSUE THREAD

**Issue by jeffbolznv** [KhronosGroup/glslang#1789]
SPIR-V doesn't have any way to set OutputVertices to a specialized value, so I assume it should be an error to set max_vertices to a specialized constant. glslang currently emits an OutputVertices of zero, which is surely wrong.

```
#version 450
layout(constant_id = 1) const uint C = 3;

layout(triangles) in;
layout(triangle_strip, max_vertices = C) out;

void main() {
}
```

```
               OpCapability Geometry
          %1 = OpExtInstImport "GLSL.std.450"
               OpMemoryModel Logical GLSL450
               OpEntryPoint Geometry %main "main"
               OpExecutionMode %main Triangles
               OpExecutionMode %main Invocations 1
               OpExecutionMode %main OutputTriangleStrip
               OpExecutionMode %main OutputVertices 0
               OpSource GLSL 450
               OpName %main "main"
               OpName %C "C"
               OpDecorate %C SpecId 1
       %void = OpTypeVoid
          %3 = OpTypeFunction %void
       %uint = OpTypeInt 32 0
          %C = OpSpecConstant %uint 3
       %main = OpFunction %void None %3
          %5 = OpLabel
               OpReturn
               OpFunctionEnd
```

**Comment by johnkslang** [comment]
Yes, it looks like this should generally be an error.  From Rev. 7 of the GLSL specification:

> integer-constant-expression is defined in “Constant Expressions” as constant integral expression, with it being a compile-time error for integer-constant-expression to be a specialization constant.

**Comment by johnkslang** [comment]
When I execute your example, as a geometry shader, I get:

```
ERROR: t.geom:5: 'max_vertices' : needs a literal integer
```

Most things are explicitly checking for a literal. 

While I agree that the non-literal ones should check for non-specialization constant, I'm confused about this example already having a stronger check.

**Comment by jeffbolznv** [comment]
This seems like it was fixed with commit 5cb2fa2ad2fe693e82fd18db955ce97ea857cf80, before I filed the issue. I was probably testing with a released SDK build and not TOT glslang. Closing.
